pkg/authentication/externalidp/awsiam/aws_client_test.go

```diff
@@ -0,0 +1,74 @@
+package awsiam
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/sts"
+	"github.com/stretchr/testify/require"
+)
+
+type mockCredentialsProvider struct {
+	creds aws.Credentials
+	err   error
+}
+
+func (m mockCredentialsProvider) Retrieve(ctx context.Context) (aws.Credentials, error) {
+	return m.creds, m.err
+}
+
+func TestGeneratePresignedURL_Integration(t *testing.T) {
+	numSeconds := 360
+	validCreds := aws.Credentials{
+		AccessKeyID: "accesKey",
```

Comment:
This test actually exposes a bug, no? 
we must have 3 parameters: secret key, access key AND session token